openeo/rest/connection.py

```diff
@@ -525,6 +526,36 @@ def list_jobs(self) -> dict:
         # TODO: duplication with `user_jobs()` method
         return self.get('/jobs').json()["jobs"]
 
+    def save_process_graph(self, process_graph_id: str, process_graph: dict, **metadata) -> RESTProcessGraph:
```

Comment:
I would use `user_defined_process` in these method names instead of `process_graph` to be closer to the terminology of the official docs.

Also, a user defined process is a bit more than a process graph, most importantly a declaration of the expected parameters. I would add an explicit argument `parameters` for that.